node/src/sync_test.rs

```diff
@@ -0,0 +1,94 @@
+use reqwest::blocking::Client;
+use serde_json::{json, Value};
+use std::{
+	io::{BufRead, BufReader},
+	process::{Child, Command, Stdio},
+	thread,
+	time::{Duration, Instant},
+};
+
+const RPC_PORT: u32 = 9933;
+const TIMEOUT: Duration = Duration::from_secs(80); // It usually takes around 60 seconds to sync
+
+fn is_synced() -> bool {
+	let request_body = json!({
+		"jsonrpc": "2.0",
+		"method": "system_health",
+		"params": [],
+		"id": 1
+	});
+
+	let client = Client::new();
+	let response = match client
+		.post(format!("http://127.0.0.1:{}", RPC_PORT))
+		.header("Content-Type", "application/json")
+		.json(&request_body)
+		.send()
+	{
+		Ok(resp) => resp,
+		Err(err) => {
+			eprintln!("[rpc] Node is not ready or connection failed: {err}");
+			// By returning `false` here, we treat "no response" as "still syncing"
+			return false;
+		},
+	};
+
+	// Parse JSON or assume "still syncing" on failure
+	let response_json: Value = match response.json() {
+		Ok(json) => json,
+		Err(err) => {
+			eprintln!("[rpc] Failed to parse JSON: {err}");
+			return false;
+		},
+	};
+
+	!response_json["result"]["isSyncing"].as_bool().unwrap_or(true)
+}
+
+fn spawn_laos_warp() -> Child {
+	let laos_bin = std::path::Path::new("..").join("target").join("release").join("laos");
```

Comment:
``laos_bin`` may be substituted by 

```suggestion
	let laos_bin = std::env::current_dir()
        .expect("the directory node exists and it doesn't have restricted permissions, this shouldn't fail;qed;")
        .parent()
        .expect("the root directory of the repo exists and it doesn't have restricted permissions, this shouldn't fail;qed;")
        .join("target")
        .join("release")
        .join("laos");
```
so the path will still be valid if we move this file around in the future